corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -320,28 +377,64 @@ hqDefine("geospatial/js/case_grouping_map",[
                 let caseGroup = caseGroupsInstance.getGroupByID(caseGroupID);
                 color = caseGroup.color;
                 const marker = new mapboxgl.Marker({ color: color, draggable: false });  // eslint-disable-line no-undef
-                marker.setLngLat([caseItem.coordinates.lng, caseItem.coordinates.lat]);
+
+                const coordinates = [caseItem.coordinates.lng, caseItem.coordinates.lat];
+                marker.setLngLat(coordinates);
 
                 // Add the marker to the map
                 marker.addTo(map);
                 mapMarkers.push(marker);
+
+                let popupDiv = document.createElement("div");
+                popupDiv.setAttribute("data-bind", "template: 'select-case'");
+
+                let popup = new mapboxgl.Popup({ offset: 25, anchor: "bottom" })  // eslint-disable-line no-undef
+                    .setLngLat(coordinates)
+                    .setDOMContent(popupDiv);
+
+                marker.setPopup(popup);
+
+                const markerDiv = marker.getElement();
+                // Show popup on hover
+                markerDiv.addEventListener('mouseenter', () => marker.togglePopup());
+
+                // Hide popup if mouse leaves marker and popup
+                var addLeaveEvent = function (fromDiv, toDiv) {
+                    fromDiv.addEventListener('mouseleave', function () {
+                        setTimeout(function () {
+                            if (!$(toDiv).is(':hover')) {
+                                // mouse left toDiv as well
+                                marker.togglePopup();
+                            }
+                        }, 100);
+                    });
+                };
+                addLeaveEvent(markerDiv, popupDiv);
+                addLeaveEvent(popupDiv, markerDiv);
+                const colors = {default: color, selected: color};
+
+                const mapMarkerInstance = new mapMarkerModel(caseItem.caseId, caseItem, marker, colors);
+                $(popupDiv).koApplyBindings(mapMarkerInstance);
             }
         });
     }
 
     function uuidv4() {
         // https://stackoverflow.com/questions/105034/how-do-i-create-a-guid-uuid/2117523#2117523
         return "10000000-1000-4000-8000-100000000000".replace(/[018]/g, c =>
-            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16),
+            (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
         );
     }
 
     function caseGroupSelectModel() {
         'use strict';
         var self = {};
 
-        self.allGroups = ko.observableArray([]);
         self.allCaseGroups;
+        // allGroups and caseGroupsForTable contains the same data, but there's weird knockoutjs behaviour
+        // if we're making allGroups an observable. caseGroupsForTable is populated by setCaseGroupsForTable
+        self.allGroups = [];
```

Comment:
It was a little difficult to initially understand the difference between `allGroups` and `allCaseGroups` since their variable names makes them sound very similar. Could they perhaps have more distinct names for what they represent? A comment explaining the difference could also work.